File: JPARecipes/src/main/java/com/skilldistillery/recipes/entities/Review.java
package com.skilldistillery.recipes.entities;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Objects;

import org.hibernate.annotations.UpdateTimestamp;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

@Entity
public class Review {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;
	private String title;
	@Column(name = "date_cooked")
	private LocalDate dateCooked;
	private String remarks;
	private String difficulty;
	private Double rating;
	@Column(name = "notes_for_future")
	private String notesForFuture;
	@Column(name = "last_update")
	@UpdateTimestamp
	private LocalDateTime lastUpdate;
	private Boolean enabled;

	public Review() {
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	public LocalDate getDateCooked() {
		return dateCooked;
	}

	public void setDateCooked(LocalDate dateCooked) {
		this.dateCooked = dateCooked;
	}

	public String getRemarks() {
		return remarks;
	}

	public void setRemarks(String remarks) {
		this.remarks = remarks;
	}

	public String getDifficulty() {
		return difficulty;
	}

	public void setDifficulty(String difficulty) {
		this.difficulty = difficulty;
	}

	public Double getRating() {
		return rating;
	}

	public void setRating(Double rating) {
		this.rating = rating;
	}

	public String getNotesForFuture() {
		return notesForFuture;
	}

	public void setNotesForFuture(String notesForFuture) {
		this.notesForFuture = notesForFuture;
	}

	public LocalDateTime getLastUpdate() {
		return lastUpdate;
	}

	public void setLastUpdate(LocalDateTime lastUpdate) {
		this.lastUpdate = lastUpdate;
	}

	public Boolean getEnabled() {
		return enabled;
	}

	public void setEnabled(Boolean enabled) {
		this.enabled = enabled;
	}

	@Override
	public int hashCode() {
		return Objects.hash(id);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Review other = (Review) obj;
		return id == other.id;
	}

	@Override
	public String toString() {
		return "Review [id=" + id + ", title=" + title + ", dateCooked=" + dateCooked + ", remarks=" + remarks
				+ ", difficulty=" + difficulty + ", rating=" + rating + ", notesForFuture=" + notesForFuture
				+ ", lastUpdate=" + lastUpdate + ", enabled=" + enabled + "]";
	}

}

File: ngRefriedRecipes/src/app/models/review.ts
import { last } from "rxjs";
import { Recipe } from "./recipe";

export class Review {
  id: number;
  title: string;
  dateCooked: string;
  remarks: string;
  difficulty: string;
  rating: number;
  notesForFuture: string;
  lastUpdate: string;
  enabled: boolean;
  recipe: Recipe;

  constructor (
    id: number = 0,
    title: string = '',
    dateCooked: string = '',
    remarks: string = '',
    difficulty: string = '',
    rating: number = 0,
    notesForFuture: string = '',
    lastUpdate: string = '',
    enabled: boolean = true,
    recipe: Recipe = new Recipe (),
  ){
    this.id = id;
    this.title = title;
    this.dateCooked = dateCooked;
    this.remarks = remarks;
    this.difficulty =  difficulty;
    this.rating = rating;
    this.notesForFuture = notesForFuture;
    this.lastUpdate = lastUpdate;
    this.enabled = enabled;
    this.recipe = recipe;
  }

}
